src/vivarium/component.py

```diff
@@ -348,7 +348,7 @@ def simulation_end_priority(self) -> int:
     # Lifecycle methods #
     #####################
 
-    def __init__(self):
+    def __init__(self) -> None:
```

Comment:
[PEP-484](https://peps.python.org/pep-0484/) clarifies that __init__ should be typed as returning None:

> (Note that the return type of __init__ ought to be annotated with -> None. The reason for this is subtle. If __init__ assumed a return annotation of -> None, would that mean that an argument-less, un-annotated __init__ method should still be type-checked? Rather than leaving this ambiguous or introducing an exception to the exception, we simply say that __init__ ought to have a return annotation; the default behavior is thus the same as for other methods.)